src/main/java/com/epam/test_generator/services/CaseService.java

```diff
@@ -153,44 +152,93 @@ public CaseDTO addCaseToSuit(Long projectId, Long suitId, EditCaseDTO editCaseDT
+        caze.setName(updateCaseDTO.getName());
         caze.setLastModifiedDate(LocalDateTime.now());
 
 
         caze = caseDAO.save(caze);
 
-        CaseDTO updatedCaseDTO = caseTransformer.toDto(caze);
-        CaseUpdateDTO updatedCaseDTOwithFailedStepIds =
-            new CaseUpdateDTO(updatedCaseDTO, failedStepIds);
+        CaseDTO updatedCaseDTO = caseDTOsTransformer.toDto(caze);
+        CaseWithFailedStepsDTO caseWithFailedStepsDTOwithFailedStepIds =
+            new CaseWithFailedStepsDTO(updatedCaseDTO, failedStepIds);
 
         caseVersionDAO.save(caze);
         suitVersionDAO.save(suit);
-        return updatedCaseDTOwithFailedStepIds;
+        return caseWithFailedStepsDTOwithFailedStepIds;
+    }
+
+    /**
+     * Updates all cases to specified in list of updateCaseDTOs.
+     * Method is deprecated and better use separate methods (updateCase, removeCases and addCaseToSuit) for creating
+     * updating and deleting a case.
+     * @param projectId id of project where to update cases
+     * @param suitId id of suit where to update cases
+     * @param editCaseDTOS list of cases to update
+     * @return list {@link CaseDTO} with all changed cases
+     * @throws MethodArgumentNotValidException
+     */
+    @Deprecated
+    public List<CaseDTO> updateCases(Long projectId, long suitId, List<EditCaseDTO> editCaseDTOS)
+            throws MethodArgumentNotValidException {
+        List<CaseDTO> updatedCases = new ArrayList<>();
+        for (EditCaseDTO editCaseDTO : editCaseDTOS) {
+            switch (editCaseDTO.getAction()) {
+                case DELETE:
+                    if (editCaseDTO.getId() == null) {
+                        throw new BadRequestException("No id in caze to remove");
+                    }
+                    updatedCases.add(removeCase(projectId, suitId, editCaseDTO.getId()));
+                    break;
+                case CREATE:
+                    updatedCases.add(addCaseToSuit(projectId, suitId, editCaseDTO));
+                    break;
+                case UPDATE:
+                    if (editCaseDTO.getId() == null) {
+                        throw new BadRequestException("No id in caze to update");
+                    }
+                    UpdateCaseDTO updateCaseDTO = new UpdateCaseDTO(
+                            editCaseDTO.getDescription(),
+                            editCaseDTO.getName(),
+                            editCaseDTO.getPriority(),
+                            editCaseDTO.getStatus(),
+                            editCaseDTO.getSteps(),
+                            editCaseDTO.getComment()
+                    );
+                    CaseDTO updatedCaseDTO = updateCase(projectId, suitId, editCaseDTO.getId(), updateCaseDTO)
+                            .getUpdatedCaseDto();
+                    updatedCases.add(updatedCaseDTO);
+                    break;
+                default:
+                    throw new BadRequestException("Wrong action argument");
```

Comment:
It's better to add information about argument itself.
When you investigate an issue via logs it isn't enough to have such message.
You may do something like this:
`throw new BadRequestException(String.format("Wrong action argument: %s", editCaseDTO.getAction()));`

Reply:
Done.